### PR TITLE
tkn completion command will bail out if something unexpected is passed  

### PR DESCRIPTION
**Pull request by rupalibehera**
# Changes

- tkn completion command will bail out if something else is passed apart from expected args
- forced the number of args to be 1, passing multiple in valid options was returning 0 for eg: ( tkn completion f b)
- Modified unit test as required
issue https://github.com/tektoncd/cli/issues/1282

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Run the code checkers with `make check`
- [ ] Regenerate the manpages, docs and go formatting with `make generated`
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

### CODE REVIEW

File: pkg/cmd/completion/completion.go
Comment (piyush-garg) [inline pkg/cmd/completion/completion.go:78]: we can use `ValidArgs []string` here, we don't need default then

Comment (rupalibehera) [reply pkg/cmd/completion/completion.go:78]: Actually I was thinking if we use `Args: cobra.ExactValidArgs(1)` then we don't require the default block in the switch case and also explicit check of the len(args) before switch.

Comment (vdemeester) [reply pkg/cmd/completion/completion.go:78]: @rupalibehera what does the `ExactValidArgs` do ? aka how does it know which value are valid ?

Comment (rupalibehera) [reply pkg/cmd/completion/completion.go:78]: it checks if the input is present in the `ValidArgs: []string{"bash", "zsh", "fish", "powershell"}` and also check if the number of parameter is correct, so we don't have to explicitly check in the code.

**Comment by chmouel** [comment]
/lgtm
Looks good to me, thanks! 👍

File: pkg/cmd/completion/completion.go
Comment (vdemeester) [reply pkg/cmd/completion/completion.go:78]: Oh nice, I like that 😉

**Comment by chmouel** [comment]
FYI we usually would rather have the contributor rebasing their multiple commits as one before merging so that makes it easier to browse the github log!

**Comment by rupalibehera** [comment]
> FYI we usually would rather have the contributor rebasing their multiple commits as one before merging so that makes it easier to browse the github log!

Yep Just did that :)

**Review by vdemeester — APPROVED** [review]
/meow

**Review by piyush-garg — APPROVED** [review]
/lgtm

/hold cancel

**Comment by chmouel** [comment]
Good job @rupalibehera on your first tekton contribution  👍🏽

**Pull request merged**
